src/shared/Core/Trace2.cs

```diff
@@ -0,0 +1,316 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Serialization;
+
+namespace GitCredentialManager;
+
+/// <summary>
+/// The different event types tracked in the TRACE2 tracing
+/// system.
+/// </summary>
+public enum Event
+{
+    Version = 0,
+    Start = 1,
+    Exit = 2
+}
+
+/// <summary>
+/// Represents the application's TRACE2 tracing system, which
+/// writes to a collector service that stores the trace information.
+/// </summary>
+public interface ITrace2 : IDisposable
+{
+    /// <summary>
+    /// Time at which application began executing.
+    /// </summary>
+    long ApplicationStartTime { get; }
+
+    /// <summary>
+    /// Add a listener that will receive/queue events and write them
+    /// to the appropriate place based on event target.
+    /// </summary>
+    /// <param name="listener">The listener to add.</param>
+    void AddListener(CollectorWriter listener);
+    
+    /// <summary>
+    /// Parse TRACE2 event target
+    /// </summary>
+    /// <param name="eventTarget">
+    /// The TRACE2 event target value. Supports the following formats:
+    /// * af_unix:<socket_type>:<absolute-pathname> -  writes to Unix Domain Socket
+    /// * \\.\pipe\git-<GUID> - writes to Windows Named Pipe
```

Comment:
sorry to nit again, this value just needs to be a path to a named pipe on the local host, so "//./pipe/<anything>" is sufficient.

Sorry if I confused you earlier.  The "git-" and <guid> literals were just the values I was using in the existing daemon.

Reply:
No need to apologize - I'm glad you caught this!